tools/integration_tests/readonly/readonly_test.go

```diff
@@ -74,17 +75,23 @@ func createTestDataForReadOnlyTests(ctx context.Context, storageClient *storage.
 		filePath := path.Join(dirPath, file.filePath)
 		// Create a storage writer for the destination object
 		object := bucketHandle.Object(filePath)
-		writer := object.NewWriter(ctx)
+		writer, err := client.NewWriterWithSupportForZB(ctx, object, storageClient)
```

Comment:
Is this method used only for ZB? IMO we should create generic methods unless its too complicated.